trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/opa/OpaRequestAdapter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spi.security.opa;
+
+import io.trino.aws.proxy.spi.security.SecurityResponse;
+
+public sealed interface OpaRequestAdapter
+{
+    record OpaRequestWrapper(OpaRequest opaRequest) implements OpaRequestAdapter
```

Comment:
Could we make the constructors of `OpaRequestWrapper` and `SecurityResponseWrapper` private/protected and provide 2 static functions in `OpaRequestAdapter`: `ofOpaRequest` and `ofSecurityResponse`?